test/test_variable_unlim.jl

```diff
@@ -39,7 +39,7 @@ defDim(ds,"lon",Inf)
 defDim(ds,"lat",110)
 v = defVar(ds,"temperature",Float32,("lon","lat"))
 data = [Float32(i+j) for i = 1:100, j = 1:110]
-v[:,1] = data[:,1]
+v[1:100,1] = data[:,1]
```

Comment:
This is also breaking the API, as @Alexander-Barth mentioned in [this comment](https://github.com/Alexander-Barth/NCDatasets.jl/pull/205#issuecomment-1512595709)

Reply:
I can see how that is more annoying to write now. 

Defining e.g. `v = grow!(v, :, 1, data[:, 1])` could let this work, let DiskArrays.jl fix the chunk pattern, and let Rasters.jl add a method for that and fix the lookups at the same time.